commands/mutedrole.js

```diff
@@ -34,13 +42,35 @@ exports.command = async (message, args, database, bot) => {
       }
     }
 
+    if (message.guild.members.resolve(bot.user.id).roles.highest.comparePositionTo(role) < 0) {
+      await message.channel.send("I'm not high enough (in the role list)!");
+      return;
+    }
+
     let config = await util.getGuildConfig(message);
+    let oldRole = config.mutedRole;
     config.mutedRole = role.id;
     await util.saveGuildConfig(config);
+    let response = await message.channel.send(`Updating current mutes...`);
 
-    let response = await message.channel.send(`Updating channel overwrites...`);
+    for (mute of await database.queryAll("SELECT * FROM moderations WHERE action = 'mute' AND active = TRUE AND guildid = ?",[message.guild.id])) {
+      let member = message.guild.members.resolve(mute.userid);
+      if (member && member.roles.cache.get(oldRole)) {
+        try {
+          await member.roles.add(role.id);
+          await member.roles.remove(oldRole);
+        } catch (e) {
+          console.error("Couldn't change muted role",e);
+        }
+      }
+    }
+
```

Comment:
The entire query and for loop is not necessary if old role = new role. Also you dont set the new role if there is no old role (`member.roles.cache.get(oldRole) === false`)?